### PR TITLE
feat(datetime): mode variant

### PR DESCRIPTION
**Pull request by EmelieLitwin**
**Describe pull-request**  
Dark mode variants for datetime component. Component need refactoring/redesign but this is better than nothing since component background is white and is going to look transparent on our light primary mode variant 

**Solving issue**  
_In case of GitHub issue add # plus the number of the issue (for example #123) OR if it is Azure then AB# and number of ticket_
Fixes: [AB#3188](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/3188)

**Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants

### CODE REVIEW

File: tegel/src/components/datetime/datetime-theme-vars.scss [hunk 1]
```diff
@@ -1,14 +1,16 @@
 :root,
 .sdds-theme-light {
   --sdds-datetime-color: var(--sdds-grey-868);
-  --sdds-datetime-background: var(--sdds-white);
+  --sdds-datetime-background-primary: var(--sdds-grey-50);
+  --sdds-datetime-background-secondary: var(--sdds-white);
+  --sdds-datetime-background: var( --sdds-datetime-background-primary);
   --sdds-datetime-icon: var(--sdds-grey-868);
```
Comment (nathalielindqvist) [inline tegel/src/components/datetime/datetime-theme-vars.scss:7]: The icon seems to be Grey958 in light mode in Figma
image

Icon also changes color to red when put in error state in the design, is there a reason as to why that is not the case in the component? :relaxed:
image

File: tegel/src/components/datetime/datetime.stories.tsx [hunk 2]
```diff
@@ -110,6 +121,7 @@ export default {
     labelText: 'Label text',
     helper: true,
```
Comment (nathalielindqvist) [inline tegel/src/components/datetime/datetime.stories.tsx:122]: There is no help text in the design, only when in error state. Should we still have a general helper text in the code? Has there been a discussion about it that is not reflected in Figma?

File: tegel/src/components/datetime/datetime-theme-vars.scss [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
   --sdds-datetime-border-bottom: var(--sdds-grey-400);
   --sdds-datetime-placeholder: var(--sdds-grey-500);
   --sdds-datetime-border-bottom-hover: var(--sdds-grey-600);
```
Comment (nathalielindqvist) [inline tegel/src/components/datetime/datetime-theme-vars.scss:11]: Might not be part of this PR but the bottom border in success state seems to be Grey958 in light mode in Figma
image
image

File: tegel/src/components/datetime/datetime.stories.tsx [hunk 4]
```diff
@@ -98,6 +98,17 @@ export default {
...
     },
+    modeVariant: {
+      name: 'Mode variant',
+      description: 'Mode variant of the component.',
+      control: {
+        type: 'radio',
+      },
+      options: ['Inherit from parent', 'Primary', 'Secondary'],
+      table: {
+        defaultValue: { summary: 'Inherit from parent' },
```
Comment (nathalielindqvist) [inline tegel/src/components/datetime/datetime.stories.tsx:109]: Should we have default values for the rest of the controls as well?

File: tegel/src/components/datetime/datetime-theme-vars.scss [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
 
   // Disabled
-  --sdds-datetime-background-disabled: var(--sdds-white);
+  --sdds-datetime-background-disabled: var(--sdds-grey-50);
   --sdds-datetime-color-disabled: var(--sdds-grey-400);
   --sdds-datetime-placeholder-disabled: var(--sdds-grey-400);
   --sdds-datetime-label-disabled: var(--sdds-grey-400);
```
Comment (nathalielindqvist) [inline tegel/src/components/datetime/datetime-theme-vars.scss:16]: Might not be part of this PR, but the label in disabled state seems to use the wrong color token, in Figma it's Grey400 in light mode and Grey700 in dark mode.
image
image

image
image

**Review by mJarsater — COMMENTED** [review]
Except for the issues raised by Nathalie this one look good to me!

File: tegel/src/components/datetime/datetime.stories.tsx (hunk 2 above)
Comment (EmelieLitwin) [reply tegel/src/components/datetime/datetime.stories.tsx:122]: > There is no help text in the design, only when in error state. Should we still have a general helper text in the code? Has there been a discussion about it that is not reflected in Figma?

Good find. Without a little research on best practice I'm not sure. I definitely think we should keep the prop, but probably add it to Figma as well. At least for success state...
